src/charm.py

```diff
@@ -863,6 +863,13 @@ def _get_configs_per_app(self, relation: Relation) -> Dict[str, Any]:
 
         return config
 
+    def _get_san_for_relation(self, relation):
+        if self._routing_mode is _RoutingMode.path:
+            return self.external_host
+        else:  # _RoutingMode.subdomain
+            prefix = "somehow, from reldata"
```

Comment:
@PietroPasotti would you be able to help out obtaining the prefix using the new schema classes?
I imagine [#269](https://github.com/canonical/traefik-k8s-operator/pull/269) may have an impact on this?